build/utils.go

```diff
@@ -67,11 +69,19 @@ func toBuildkitExtraHosts(inp []string, mobyDriver bool) (string, error) {
 		if !ok || host == "" || ip == "" {
 			return "", errors.Errorf("invalid host %s", h)
 		}
-		// Skip IP address validation for "host-gateway" string with moby driver
-		if !mobyDriver || ip != mobyHostGatewayName {
-			if net.ParseIP(ip) == nil {
-				return "", errors.Errorf("invalid host %s", h)
+		// If the IP Address is a "host-gateway", replace this value with the
+		// IP address provided by the worker's label.
+		if ip == mobyHostGatewayName {
+			hgip, err := nodeDriver.HostGatewayIP(ctx)
+			if err != nil {
+				return "", err
+			} else if hgip == nil {
+				return "", errors.Errorf("unable to derive the IP value for host-gateway")
 			}
```

Comment:
nit: don't think we need an "else" here (first one already returns). Also can use `errors.New()` here, because there's no formatting 😅 

```suggestion
			if err != nil {
				return "", err
			}
			if hgip == nil {
				return "", errors.New("unable to derive the IP value for host-gateway")
			}
```

Reply:
Also left a comment on the `hgip == nil` case (ideally it would _either_ return an error _or_ return `nil`, not both (I ... think?).

Reply:
Yeah returning an error is better you right, don't need an extra cond on this.